### ISSUE TITLE
Seed Search Crash

### ISSUE THREAD

**Issue by rlvandaveer** [Cubitect/cubiomes-viewer#290]
I am sort of loathe to open this issue as I know I'm on an unsupported platform but I'd thought I'd take a stab to see if I maybe have a stupid problem. I have built and run this application on my Mac Mini many times. I had been running 3.3.0, which I built myself sometime last year. Yesterday I decided to get latest and build fresh. All seems to work well except that I can no longer search for seeds. The application crashes every time.

I started troubleshooting modestly by wiping the build folder but that didn't help. I also started over - recloning the application and rebuilt. Nada. I tried to go back to a variety of different tags but stopped at 3.3.0 e.g, `git checkout tags/3.3.0`. I couldn't get the project to build for anything but latest. I suspect that checking out a tag doesn't get the right version of the submodule and I don't know how to specify that. Any assistance would be appreciated (but not expected.) Thanks.

[crash-report.txt](https://github.com/Cubitect/cubiomes-viewer/files/14860057/crash-report.txt)

**Comment by JakobDev** [comment]
> I suspect that checking out a tag doesn't get the right version of the submodule

You can download a tarball (not the one generated by GitHub) on the Release page which includes the full source.

**Comment by rlvandaveer** [comment]
Good idea, but alas that does not work. The tarball does not include the cubiomes submodule source.

**Comment by rlvandaveer** [comment]
Regarding compiling earlier versions, I was finally able to compile `3.3.0` and `4.0.0` by properly checking out the earlier tag. I'm not very experienced with submodules and just needed to specify the `--recurse-submodules` parameter during `checkout`. N00b problem. (Interestingly enough, I couldn't get `3.4.1` or `3.4.2` to compile. I received errors with `stdlib.h not found`.)

Note that I can search for seeds in `4.0.0`. The crash reported above does not happen.

**Comment by JakobDev** [comment]
> The tarball does not include the cubiomes submodule source.

As I said, don't use the Tarball generated by GitHub. Use the manually uploaded `-src.tar.gz` tarball instead.
